### PR TITLE
Small types & factories fixes/updates

### PR DESCRIPTION
**Pull request by DevDW**
## Description
Fix TypeScript error from `id` field in `databaseBackupFactory` and update `databaseType` interface and `databaseTypeFactory` per API specs

### CODE REVIEW

**Review by bnussman — CHANGES_REQUESTED** [review]
I few things I found we should change: 
- The `DatabaseFactory` should now return `failover_count`
- `getEngineDatabase` should return `CreateDatabaseResponse` type instead of `Database`
- It would be nice to have two factories for database instances since `getEngineDatabase` and `getDatabases` return different schemes for database instances

File: packages/api-v4/src/databases/types.ts [hunk 1]
```diff
@@ -87,17 +89,18 @@ interface ConnectionStrings {
...
   label: string;
   region: string;
   status: DatabaseStatus;
-  type: string;
-  standby_count: StandbyCount;
+  type: DatabaseType;
+  failover_count: FailoverCount;
   engine: Engine;
   encrypted: boolean;
   ipv4_public: string[];
```
Comment (HanaXu) [inline packages/api-v4/src/databases/types.ts:103]: is `ipv4_public` supposed to be a string or string array?

Comment (bnussman) [reply packages/api-v4/src/databases/types.ts]: `string` I believe

Comment (DevDW) [reply packages/api-v4/src/databases/types.ts]: looks like a string -- just fixed it, thanks for catching

**Review by bnussman — APPROVED** [review]
Changes looks good. I like the decision to use `DatabaseInstance` and `Database` 🚀

**Review by HanaXu — APPROVED** [review]
🚀

**Review by bcmertz — APPROVED** [review]
LGTM!

**Pull request merged**